src/cdcacm_serial.c

```diff
@@ -89,9 +89,7 @@ static void write_data(struct device *dev, const char *buf, int len)
 	uart_fifo_fill(dev, (const uint8_t*)buf, len);
 	while (data_transmitted == false)
```

Comment:
I actually noticed that having the yield it gets stuck inside the while loop. 
The originall cdc-acm example from Zephyr does not have the yield in there.